docker/service.go

```diff
@@ -460,31 +462,33 @@ func (s *Service) recreateIfNeeded(ctx context.Context, c *Container, noRecreate
 	return c, err
 }
 
-func (s *Service) recreate(ctx context.Context, c *Container) (*Container, error) {
+func (s *Service) recreate(ctx context.Context, c *container.Container) (*container.Container, error) {
 	name := c.Name()
-	newName := fmt.Sprintf("%s_%s", name, c.container.ID[:12])
+	id, _ := c.ID()
```

Comment:
Can `ID` be changed to not return an error? It seems to always be `nil` anyways.